include/spi.h

```diff
@@ -27,173 +37,184 @@
 extern "C" {
 #endif
 
+/**x
+ * @brief SPI operational mode
+ */
+#define SPI_OP_MODE_MASTER	0
+#define SPI_OP_MODE_SLAVE	1
+
 /**
  * @brief SPI Polarity & Phase Modes
  */
-#define SPI_MODE_CPOL		0x1
-#define SPI_MODE_CPHA		0x2
-#define SPI_MODE_LOOP		0x4
+#define SPI_MODE_CPOL		BIT(1)
+#define SPI_MODE_CPHA		BIT(2)
+#define SPI_MODE_LOOP		BIT(3)
 
-#define SPI_MODE_MASK		(0x7)
-#define SPI_MODE(_in_)		((_in_) & SPI_MODE_MASK)
+#define SPI_MODE_MASK		(0xE)
+#define SPI_MODE_GET(_mode_)			\
+	((_mode_) & SPI_MODE_MASK)
 
 /**
  * @brief SPI Transfer modes (host controller dependent)
  */
-#define SPI_TRANSFER_MSB	(0 << 3)
-#define SPI_TRANSFER_LSB	(1 << 3)
+#define SPI_TRANSFER_MSB	(0)
+#define SPI_TRANSFER_LSB	BIT(4)
 
-#define SPI_TRANSFER_MASK	(0x8)
+#define SPI_TRANSFER_MASK	(0x10)
 
-#define SPI_WORD_SIZE_MASK	(0xFF << 4)
-#define SPI_WORD_SIZE_GET(_in_) (((_in_) & SPI_WORD_SIZE_MASK) >> 4)
-#define SPI_WORD(_in_) ((_in_) << 4)
+/**
+ * @brief SPI word size
+ */
+#define SPI_WORD_SIZE_MASK	(0x3F << 5)
```

Comment:
True, I actually used that way in my spi_update_wip branch (it's about supporting some features in dw driver).